app.js

```diff
@@ -119,10 +122,44 @@ app.use(function(req, res, next) {
 });
 
 /**
- * Last chance: is this a static asset?
+ * Is this a static asset?
  */
 app.use(express.static(DIST_DIR));
 
+
+/**
+ * Last chance: check if it's a WP page, otherwise send 404
+ */
+app.use(function(req, res, next) {
+  require('babel-register')({
+    extensions: [".jsx", ".js"],
+    presets: ['react']
+  });
+  var ReactDOMServer = require('react-dom/server');
+  var Page = require('./components/page.jsx');
+  var HtmlWrapper = require('./components/HTML-wrapper.jsx');
+  var WpContentLoader = require('react-wp-content-loader');
+
+  wpPageChecker(req.path, function(err, wpContent) {
+    console.log("\n\n\n\n Page", Page);
+
+    if ( err ) {
+      res.status(404).send(notFoundHTML);
+    } else {
+      var PageContent = React.createElement(Page, 
+        {
+          routes: [
+            { path: '', component: { pageClassName: '', pageTitle: ''} }
+          ] 
+        }, 
+        React.createElement('div', { dangerouslySetInnerHTML: { __html: wpContent }})
+      );
+
+      res.status(200).send("<!DOCTYPE html>" + ReactDOMServer.renderToStaticMarkup(PageContent));
```

Comment:
(Note that `PageContent` will need to be wrapped in HTML/Page wrapper still, e.g., `<html><head>...</head><body>...</body></html>`)

1. run `npm run app`
2. visit `http://localhost:8008/new-page` (`new-page` is a slug exists in the WP instance)
3. `ReactDOMServer.renderToStaticMarkup` throws an error as below
```bash
Warning: React.createElement: type should not be null, undefined, boolean, or number. It should be a string (for DOM elements) or a ReactClass (for composite components). Check the render method of `Page`.

/Users/mavisou/Development/teach.webmaker.org/node_modules/fbjs/lib/invariant.js:39
      error = new Error(format.replace(/%s/g, function () {
              ^
Invariant Violation: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object. Check the render method of `Page`.
    at invariant (/Users/mavisou/Development/teach.webmaker.org/node_modules/fbjs/lib/invariant.js:39:15)
    at instantiateReactComponent (/Users/mavisou/Development/teach.webmaker.org/node_modules/react/lib/instantiateReactComponent.js:64:134)
    at instantiateChild (/Users/mavisou/Development/teach.webmaker.org/node_modules/react/lib/ReactChildReconciler.js:29:28)
    at traverseAllChildrenImpl (/Users/mavisou/Development/teach.webmaker.org/node_modules/react/lib/traverseAllChildren.js:98:5)
    at traverseAllChildrenImpl (/Users/mavisou/Development/teach.webmaker.org/node_modules/react/lib/traverseAllChildren.js:114:23)
    at traverseAllChildren (/Users/mavisou/Development/teach.webmaker.org/node_modules/react/lib/traverseAllChildren.js:186:10)
    at Object.ReactChildReconciler.instantiateChildren (/Users/mavisou/Development/teach.webmaker.org/node_modules/react/lib/ReactChildReconciler.js:52:5)
    at ReactDOMComponent.ReactMultiChild.Mixin._reconcilerInstantiateChildren (/Users/mavisou/Development/teach.webmaker.org/node_modules/react/lib/ReactMultiChild.js:197:41)
    at ReactDOMComponent.ReactMultiChild.Mixin.mountChildren (/Users/mavisou/Development/teach.webmaker.org/node_modules/react/lib/ReactMultiChild.js:232:27)
    at ReactDOMComponent.Mixin._createContentMarkup (/Users/mavisou/Development/teach.webmaker.org/node_modules/react/lib/ReactDOMComponent.js:591:32)
```
I found out the error came from our custom React components used in `Page.jsx` (e.g., `<Sidebar>`, `<Footer>` etc). If we remove those components from the Page's render then the errors are gone. I'm not exactly sure why... `babel-preset-react` is there and if it's complaining about `jsx` wouldn't we hit error at the first "<" occurrence in `Page.jsx`?

@Pomax any pointers? :cold_sweat: